firedantic/_async/model.py

```diff
@@ -94,7 +102,9 @@ def _cls(doc_id: str, data: Dict[str, Any]) -> TAsyncBareModel:
                     data[cls.__document_id__],
                 )
             data[cls.__document_id__] = doc_id
-            return cls(**data)
+            model = cls(**data)
```

Comment:
This was apparently necessary to fully support `_doc_id` type values for `__document_id__`